tools/c7n_azure/tests/test_cosmos_db.py

```diff
@@ -502,7 +510,13 @@ def test_restore_throughput_state_updates_throughput_from_tag(self):
         self._assert_offer_throughput_equals(throughput_to_restore, collections[0]['_self'])
 
     def _assert_offer_throughput_equals(self, throughput, resource_self):
+        live_session_wait()
         offers = self.data_client.ReadOffers()
         offer = next((o for o in offers if o['resource'] == resource_self), None)
         self.assertIsNotNone(offer)
-        self.assertEqual(offer['content']['offerThroughput'], throughput)
+        self.assertEqual(throughput, offer['content']['offerThroughput'])
+
+
+def live_session_wait():
```

Comment:
I was thinking we could add this to `BaseTest` in `azure_common.py` - most methods there already check `if self.is_playback():` so we could have `def test_sleep(self, interval)` that does nothing if `self.is_playback()` - then it'll run during cassette recording or live tests only.

Reply:
I named the functions `sleep_in_live_mode`. I tried `test_sleep` but then it got picked up as a test for each running class, which added a ton of time